### DISCUSSION TITLE
dispose of temp-files when using StreamingXlsx in spudsoft XlsxEmitter

### DISCUSSION THREAD

**Discussion by aremund** [eclipse-birt/birt#1532] [Q&A] [answered]
We generate large xlsx-Files and want to avoid high memory-consumption. Therefore, we make use of the streaming-option `ExcelEmitter.StreamingXlsx` introduced with https://github.com/eclipse-birt/birt/issues/1322 This works great!

Since using this streaming-option, we get temporary files generated in `/tmp/poifiles` We did a bit of research and the Apache POI streaming-extension achieves the reduction of memory by writing those temp-files. In particular, I found the following remark https://poi.apache.org/components/spreadsheet/how-to.html#sxssf
> Note that SXSSF allocates temporary files that you **must** always clean up explicitly, by calling the dispose method.

I looked at the implementation of the option `ExcelEmitter.StreamingXlsx` and did not find any such call. Would it be possible, to add such a call to tidy up the temporary files?
In the meantime, we will just delete those temp-files on a regular basis.

**Comment by hvbtup** [comment]
Probably the disposal should take place after the emitter completed writing to the output stream.
Please create a Pull Request for this.

**Comment by speckyspooky** [comment, marked as answer]
Will be imrpived through PR #1535
